src/k_means_cluster.js

```diff
@@ -0,0 +1,123 @@
+import sample from "./sample";
+
+/**
+ * Perform k-means clustering.
+ *
+ * @param {Array<Array<number>>} points XY coordinates of points to cluster.
```

Comment:
Afaict, this is n-dimensional clustering, so referring to points as XY is too limiting?

Reply:
Thanks - fixed.